\documentclass[portrait,a0,final]{a0poster}

\usepackage{epsf,pstricks}
\usepackage[utf8]{inputenc}
\usepackage[sc]{mathpazo}
\usepackage[T1]{fontenc}
%\usepackage{hyperref}
\usepackage{geometry}
\geometry{verbose,tmargin=1.0cm,bmargin=1.5cm,lmargin=1.5cm,rmargin=1.5cm}
\setcounter{secnumdepth}{2}
\setcounter{tocdepth}{2}
\usepackage{url}
\usepackage[unicode=true,pdfusetitle,
 bookmarks=true,bookmarksnumbered=true,bookmarksopen=true,bookmarksopenlevel=2,
 breaklinks=ning,pdfborder={0 0 1},backref=ning,colorlinks=ning]
 {hyperref}
\hypersetup{pdfstartview={XYZ null null 1}}
\usepackage{authblk}
\usepackage{nopageno}
\renewcommand{\familydefault}{\rmdefault}

% For tikz
\usepackage{tikz}
\usetikzlibrary{shapes,arrows,snakes}
\usepackage{amsmath,amssymb}
\usetikzlibrary{positioning}

\title{\Huge MetaboDiff: an R package for differential metabolomic analysis.}
\date{}
\author{{\Large Andreas Mock and Christel Herold-Mende}\\ \normalsize Division of Experimental Neurosurgery, Department of Neurosugery, Heidelberg University Hospital}


\begin{document}
\pagestyle{empty}
\maketitle
\Large


% Define block styles
\tikzstyle{decision} = [diamond, draw, fill=blue!20, text width=4.5em, text badly centered, node distance=3cm, inner sep=0pt]
\tikzstyle{block} = [rectangle, draw, fill=blue!20, text width=5em, text centered, rounded corners, minimum height=4em]
\tikzstyle{greenbox} = [rectangle, draw=blue, fill=green!20, text width=5em, text centered, rounded corners, inner sep=10pt, inner ysep=12pt, very thick]
\tikzstyle{line} = [draw, -latex']
\tikzstyle{cloud} = [draw, ellipse, node distance=4cm, minimum height=2em, text width=5em]
\tikzstyle{mybox} = [draw=blue, fill=green!20, very thick, rectangle, rounded corners, inner sep=10pt, inner ysep=20pt]
\tikzstyle{myboxwhite} = [rectangle, rounded corners, inner sep=10pt, inner ysep=0pt]
\tikzstyle{myboxblue} = [rectangle, fill=blue!10, rounded corners, inner sep=10pt, inner ysep=20pt]
\tikzstyle{fancytitle} = [fill=white, text=black, ellipse, draw=blue]


\vspace{0cm}


\begin{tikzpicture}[every node/.style = {anchor=south}]]


\node [myboxwhite] (intro){%
    \begin{minipage}{0.45\textwidth}

{\bf Introduction} Comparative metabolomics comes of age by an increasing list of commercial vendors (i.e. Metabolon\textregistered) offering reproducible high-quality metabolomic data for translational researchers outside the mass spectrometry field. This R packages aims to provide a low-level entry to differential metabolomic analysis by starting off with the table of relative metabolite quantifications provided by commercial vendors.

    \end{minipage}

};


\node [myboxwhite,  below=2.5cm of intro] (installation){%
    \begin{minipage}{0.45\textwidth}

{\bf Installation of \emph{R} package}

<<eval=TRUE, echo=TRUE, message=FALSE>>=
library(devtools)
install_github("andreasmock/MetaboDiff")
@

The package vignette can be found on Github at \\  \textcolor{blue}{\url{https://github.com/andreasmock/MetaboDiff}}.

    \end{minipage}

};

\node [myboxblue, below=2.5cm of installation] (example){%
    \begin{minipage}{0.45\textwidth}

{\bf Example data} The example data is derived from a study by Priolo and colleagues in which they used the service of the Metabolon company to compare the tissue metabolome of 40 prostate cancers with 16 normal prostate specimens.
  \end{minipage}

};

\node [myboxwhite, below=2.5cm of example] (mae){%
    \begin{minipage}{0.45\textwidth}

{\bf Data representation} The metabolomic data within \verb|MetaboDiff| is stored as a \verb|MultiAssayExperiment| class (Sig, 2015). This framework enables the coordinated representation of multiple experiments on partially overlapping samples with associated metadata and integrated subsetting across experiments. In the context of metabolomic data analysis, multiple assays are needed to store raw data and imputed data. \\
\\
The core components of the \verb|MultiAssayExperiment| class are:

\begin{itemize}

\item \verb|ExperimentList| - a slot of class ExperimentList containing data for each experimental assay. Within the ExperimentList slot, the metabolomic data is stored as a 
SummarizedExperiment object consisting of:
    \subitem \verb|assay| - a matrix containing the relative measurements. 
    \subitem \verb|rowData| - a dataframe containing the metabolite annotation.
\item \verb|colData| - a slot of class data frame describing the sample metadata available across all experiments.
\item \verb|sampleMap| - a slot of class data frame relating clinical data to experimental assay.

\end{itemize}

    \end{minipage}

};

\node [myboxwhite,  below=2.5cm of mae] (imputation){%
    \begin{minipage}{0.45\textwidth}

{\bf Imputation of missing values} In contrast to microarrays, missing values are common in quantitative metabolomic datasets. The following heatmap shows the missing values across the example data. K-nearest neighbor imputation was used to minimize effects on the normality and variance of the data (Armitage et al., 2015).

<<eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE>>=
library(devtools)
install_github("andreasmock/MetaboDiff")
library("devtools")
library("gdata")
library("MultiAssayExperiment")
library("SummarizedExperiment")
library("MetaboDiff")
library("tidyverse")
library("ggfortify")
library("ComplexHeatmap")
library("impute")
library("vsn")
require("cowplot")
library("ggbiplot")
@


<<eval=TRUE,echo=FALSE,message=FALSE,warning=FALSE,cache=TRUE>>=
data = as.matrix(read.xls("http://cancerres.aacrjournals.org/highwire/filestream/290852/field_highwire_adjunct_files/4/131853_1_supp_2658512_nbpsn6.xlsx",sheet=5,na.strings = ""))

assay = apply(data[3:nrow(data),8:ncol(data)],2,as.numeric)
colnames(assay) = paste0(rep("sample",ncol(assay)),1:ncol(assay))
rownames(assay) = paste0(rep("met",nrow(assay)),1:nrow(assay))


colData = data.frame(id = colnames(t(data[1,8:ncol(data)])),
                     tumor_normal = as.vector(t(data[1,8:ncol(data)])),
                   row.names=paste0(rep("pat",ncol(assay)),1:ncol(assay)))


rowData = as.data.frame(data[3:nrow(data),1:7])
colnames(rowData) = data[2,1:7]
colnames(rowData)[7] = "HMDB_ID"
rownames(rowData) = paste0(rep("met",nrow(assay)),1:nrow(assay))


rowData = get_SMPDBanno(rowData,
                        column_kegg_id=6,
                        column_hmdb_id=7,
                        column_chebi_id=NA)

se = SummarizedExperiment(assays=assay,
                           rowData=rowData)
experiment_list = list(raw=se)


sampleMap = data.frame(primary=rownames(colData),
                 colname=colnames(se))
sampleMap_list = listToMap(list(raw=sampleMap))

met = MultiAssayExperiment(experiments = experiment_list,
                           colData = colData,
                           sampleMap = sampleMap_list)

met = knn_impute(met,cutoff=0.4)

met = normalize_met(met)
@

\vspace{1cm}

<<message=FALSE,warning=FALSE,fig.width=13,fig.height=5,echo=FALSE,fig.align="center">>=
na_heatmap(met,
           sample_label=colData(met)$tumor_normal,
           label_colors=c("darkseagreen","dodgerblue"))
@

    \end{minipage}

};

\node [myboxblue, below=2.5cm of imputation] (about){%
    \begin{minipage}{0.45\textwidth}

{\bf About this poster} This poster was created using a R markdown template of rOpenGov (\textcolor{blue}{\url{http://ropengov.github.io}}). It is fully reproducible; the full source code of this poster is available at \textcolor{blue}{\url{https://github.com/andreasmock/MetaboDiff_poster}}.
  \end{minipage}
};

\node [myboxwhite, right=3cm of intro, yshift=-6.9cm] (norm){%
    \begin{minipage}{0.45\textwidth}

{\bf Normalization} Variance stabilizing normalization (vsn) was originally developed for microarrys to ensure that the variance remains nearly constant over the whole intensity spectrum (Huber et al., 2002). It could be shown that vsn normalization performs also very well for metabolomic data (Kohl et al., 2012). The following plots illustrate the variance stabilization before (A) and after normalization (B) in the example data:

\vspace{1cm}

<<message=FALSE,warning=FALSE,fig.width=11,fig.height=5,echo=FALSE,fig.align="center">>=
df1 = data.frame(mean=apply(assays(met)[["norm"]],1,mean,na.rm=TRUE),
                sd=apply(assays(met)[["norm"]],1,sd,na.rm=TRUE))
plot1 = ggplot(df1, aes(x=mean, y=sd)) + geom_point() + geom_smooth(method=lm)  

df2 = data.frame(mean=apply(log2(assays(met)[["raw"]]),1,mean,na.rm=TRUE),
                sd=apply(log2(assays(met)[["raw"]]),1,sd,na.rm=TRUE))
plot2 = ggplot(df2, aes(x=mean, y=sd)) + geom_point() + geom_smooth(method=lm)  

plot_grid(plot2, plot1, align='h', labels=c('A', 'B'),label_size = 24)
@


    \end{minipage}

};

\node [myboxwhite, below=2.5cm of norm] (unsup){%
    \begin{minipage}{0.45\textwidth}

{\bf Unsupervised analysis} A number of unsupervised analysis and visualizations are at offer within \verb|MetaboDiff| including correlation heatmaps and PCA plots: 

\vspace{1.5cm}

<<message=FALSE,warning=FALSE,fig.width=11,fig.height=5,echo=FALSE,fig.align="center">>=
library(circlize)
library(RColorBrewer)

data = t(assays(met)[["norm_imputed"]])
data.pca=prcomp(data,scale. = TRUE)
plot1 = ggbiplot(data.pca,groups=colData(met)$tumor_normal,ellipse=TRUE,var.axes = FALSE) + 
scale_color_manual(name="grouping", values=c("darkseagreen","dodgerblue")) + xlab("PC1 (18.9% explained var.)") + ylab("PC2 (10.2% explained var.)")


mat = cor(t(assays(met)[["norm_imputed"]]))
hc = hclust(as.dist(1 - mat))
mat = mat[hc$order, hc$order]
mat[lower.tri(mat)] = 0

col_fun = colorRamp2(c(-0.5, 0, 0.5), c("#67A9CF", "white", "#EF8A62"), transparency = 0)
group_color = structure(brewer.pal(9,"Set3"), names = levels(rowData(met[["norm_imputed"]])$SUPER_PATHWAY))

col_anno = columnAnnotation(df=data.frame(pathway=rowData(met[["norm_imputed"]])$SUPER_PATHWAY),col=list(pathway=group_color),
                            show_legend = FALSE)

row_anno = rowAnnotation(df=data.frame(pathway=rowData(met[["norm_imputed"]])$SUPER_PATHWAY),col=list(pathway=group_color))

Heatmap(mat,cluster_rows = FALSE,
        cluster_columns=FALSE, 
        show_column_names = FALSE,
        show_row_names = FALSE,
        name="PCC", 
        top_annotation = col_anno,
        show_row_dend = FALSE,
        col=col_fun) + row_anno

vp.BottomLeft <- viewport(height=unit(.8, "npc"), width=unit(.8, "npc"), 
                           just=c("left","bottom"), 
                           y=-0.02, x=-0.2)

print(plot1,vp=vp.BottomLeft)

@


    \end{minipage}

};

\node [myboxwhite, below=2.5cm of unsup] (sup){%
    \begin{minipage}{0.45\textwidth}

{\bf Differential analysis} Metabolites are compared using Student T-Tests. Correction for multiple testing is performed by independent hypothesis weighting (IWH; Ignatiadis et al., 2016) with variance as a covariate. Differential pathways are identified by enrichment analyses.

\vspace{1cm}


<<message=FALSE,warning=FALSE,fig.width=11,fig.height=6,echo=FALSE,fig.align="center",error=FALSE>>=
layout(matrix(c(1,1,2),1,3))
library(genefilter)
library(IHW)
library(beeswarm)
df = rowttests(assays(met)[["norm_imputed"]],
          fac = as.factor(colData(met)$tumor_normal))
df_ihw = as.data.frame(ihw(df$p.value,apply(assays(met)[["norm_imputed"]],1,var),alpha = 0.05))
plot(df$dm, -1*log10(df_ihw$adj_pvalue),bty="n",pch=20,
     xlab="log2 fold-change",ylab="adjusted -log10 p-value")
abline(h=-log10(0.05),lty=2)
abline(v=log2(1.5),lty=2)
abline(v=-log2(1.5),lty=2)
Tsig = df$dm<(-log2(1.5))&df_ihw$adj_pvalue<0.05
points(x = df$dm[Tsig],y=-1*log10(df_ihw$adj_pvalue)[Tsig],col="darkseagreen",pch=20)
Nsig = df$dm>(log2(1.5))&df_ihw$adj_pvalue<0.05
points(x = df$dm[Nsig],y=-1*log10(df_ihw$adj_pvalue)[Nsig],col="dodgerblue",pch=20)
points(x=c(-1.201373621,max(df$dm)),y=c(2.696225,2.696225),type="l",col="brown")
text(x=-1.151373621,y=2.8,labels = "guanosine",cex = 0.7)

df = data.frame(value=assays(met)[["norm"]][134,],
                tumor_normal=colData(met)$tumor_normal)
boxplot(df$value~df$tumor_normal,
        frame=FALSE,ylab="normalized abundance of guanosine",
        col=c("darkseagreen","dodgerblue"))
box("plot",col="brown")
@


    \end{minipage}

};

\node [myboxblue, below=2.5cm of sup] (todo){%
    \begin{minipage}{0.45\textwidth}

{\bf To do} 
\begin{itemize}
\item Implementation of HotNet2 algorithm to identify significantly altered subpathways (Pleiserson et. al, 2014)
\item Hive plot of metabolic network (Krzywinski et al., 2012)
\end{itemize}

  \end{minipage}

};

\node [myboxwhite, below=2.5cm of todo] (references){%
    \begin{minipage}{0.45\textwidth}

    \small{
    \section*{References}

    \begin{enumerate}
      \item Armitage, EG et al. (2015). Missing value imputation strategies for metabolomics data. Electrophoresis, 36(24), 3050–3060.
    \item Huber, W et al. (2002). Variance stabilization applied to microarray data calibration and to the quantification of differential expression. Bioinformatics, 18 Suppl 1, S96–104.
    \item Ignatiadis, N et al. (2016). Data-driven hypothesis weighting increases detection power in genome-scale multiple testing. Nature Methods, 13(7), 577–580. 
    \item Kohl, SM et al. (2012). State-of-the art data normalization methods improve NMR-based metabolomic analysis. Metabolomics, 8(Suppl 1), 146–160. 
    \item Krzywinski, M et al. (2012). Hive plots--rational approach to visualizing networks. Briefings in Bioinformatics, 13(5), 627–644.
    \item PLeiserson, MDM et al. (2014). Pan-cancer network analysis identifies combinations of rare somatic mutations across pathways and protein complexes. Nature Genetics, 47(2), 106–114.
    \item Sig, M (2017). MultiAssayExperiment: Software for the integration of multi-omics experiments in Bioconductor. R package version 1.2.1
    \end{enumerate}
}
    \end{minipage}
};

\end{tikzpicture}

\end{document}